src/QMCWaveFunctions/TrialWaveFunction.cpp

```diff
@@ -330,18 +336,45 @@ TrialWaveFunction::RealType TrialWaveFunction::ratio(ParticleSet& P, int iat)
 TrialWaveFunction::ValueType TrialWaveFunction::calcRatio(ParticleSet& P, int iat, ComputeType ct)
 {
   ValueType r(1.0);
-  std::vector<WaveFunctionComponent*>::iterator it(Z.begin());
-  std::vector<WaveFunctionComponent*>::iterator it_end(Z.end());
-  for (int ii = V_TIMER; it != it_end; ++it, ii += TIMER_SKIP)
+  for (int i = 0, ii = V_TIMER; i < Z.size(); i++, ii += TIMER_SKIP)
   {
     myTimers[ii]->start();
-    if (ct == ComputeType::ALL || ((*it)->is_fermionic && ct == ComputeType::FERMIONIC) || (!(*it)->is_fermionic && ct == ComputeType::NONFERMIONIC))
-      r *= (*it)->ratio(P, iat);
+    if (ct == ComputeType::ALL || (Z[i]->is_fermionic && ct == ComputeType::FERMIONIC) ||
+        (!Z[i]->is_fermionic && ct == ComputeType::NONFERMIONIC))
+      r *= Z[i]->ratio(P, iat);
     myTimers[ii]->stop();
   }
   return r;
 }
 
+void TrialWaveFunction::flex_calcRatio(const std::vector<TrialWaveFunction*>& WF_list,
```

Comment:
mw_ always work on the whole list with 1 algorithm.
flex_ handles dispatching to single walker or mw_ versions.
Logically yes it evaluates 1 or more wavefunction/particle pairs.
Because TrialWaveFunction and WaveFunctionComponent are at different hierarchies, I prefer to have different names at the moment. Drivers and Hamiltonians should call flex_ functions not mw_ ones. Maybe having different names are not necessary, we can change it later by `sed` easily.

Reply:
Ok I suspected something along that line.